src/partition_and_pq.cpp

```diff
@@ -379,6 +363,269 @@ int generate_pq_pivots(const float *passed_train_data, size_t num_train,
   return 0;
 }
 
+int generate_opq_pivots(const float *passed_train_data, size_t num_train,
+                        unsigned dim, unsigned num_centers,
+                        unsigned num_pq_chunks, std::string opq_pivots_path,
+                        bool make_zero_mean) {
+  if (num_pq_chunks > dim) {
+    diskann::cout << " Error: number of chunks more than dimension"
+                  << std::endl;
+    return -1;
+  }
+
+  std::unique_ptr<float[]> train_data =
+      std::make_unique<float[]>(num_train * dim);
+  std::memcpy(train_data.get(), passed_train_data,
+              num_train * dim * sizeof(float));
+
+  std::unique_ptr<float[]> rotated_train_data =
+      std::make_unique<float[]>(num_train * dim);
+  std::unique_ptr<float[]> rotated_and_quantized_train_data =
+      std::make_unique<float[]>(num_train * dim);
+
+  std::unique_ptr<float[]> full_pivot_data;
+
+  // rotation matrix for OPQ
+  std::unique_ptr<float[]> rotmat_T;
```

Comment:
variable name ending in _T is confusing as it might suggest a type. Perhaps rename to rotmat_tr (or transpose)?

Reply:
renamed it